### DISCUSSION TITLE
Get monaco editor's instance inside the extension

### DISCUSSION THREAD

**Discussion by ozair-dev** [microsoft/vscode-discussions#357] [Extension Development QnA] [answered]
Hi, Is there a way to get the monaco editor's instance inside the extension? The feature i want to add requires to have access to the editor's **model**. Also I want to be able to run editor commands (e.g, fold, unfold etc) on a specific editor. Currently, these commands are executed only on the active editor. This can be easily done in monaco editor.

**Comment by r3m0t** [comment, marked as answer]
No, the editor model is in the webpage, but the extension runs in a separate node process called the extension host.

**Reply by ozair-dev** [reply]
Is there a way to run vscode.commands.executeCommand on a specific editor rather than the active editor? I know I can make an editor active and run the command, but this makes the the current editor inactive. The command should run in background on multiple editors while the user is typing in an editor.

**Reply by ozair-dev** [reply]
The extension that I'm making basically opens multiple editors for a single document and each document only shows one function/class and hides all other code. As soon as user inserts or remove a line inside a hidden range from an other editor, the hidden region unfolds, so I have to manually fold it again. Is there a way to prevent the region from unfolding when a line is inserted or removed inside it? Adding text in the line doesn't unfold the area.
Here is a video demonstration
![209697790-695ef5ac-4914-4591-b8c1-a1083b6b64a0](https://user-images.githubusercontent.com/65649182/210153970-e1e4c33c-07dc-4383-8b68-7f26c9b05e50.gif)

**Reply by olguzzar** [reply]
Hi @ozair-dev, bumping this since it's been a few months - is this still an issue you need help with?

**Reply by ozair-dev** [reply]
> The extension that I'm making basically opens multiple editors for a single document and each document only shows one function/class and hides all other code. As soon as user inserts or remove a line inside a hidden range from an other editor, the hidden region unfolds, so I have to manually fold it again. Is there a way to prevent the region from unfolding when a line is inserted or removed inside it? Adding text in the line doesn't unfold the area. Here is a video demonstration ![209697790-695ef5ac-4914-4591-b8c1-a1083b6b64a0](https://user-images.githubusercontent.com/65649182/210153970-e1e4c33c-07dc-4383-8b68-7f26c9b05e50.gif) [ ![209697790-695ef5ac-4914-4591-b8c1-a1083b6b64a0](https://user-images.githubusercontent.com/65649182/210153970-e1e4c33c-07dc-4383-8b68-7f26c9b05e50.gif) ](https://user-images.githubusercontent.com/65649182/210153970-e1e4c33c-07dc-4383-8b68-7f26c9b05e50.gif) [ ](https://user-images.githubusercontent.com/65649182/210153970-e1e4c33c-07dc-4383-8b68-7f26c9b05e50.gif)

This issue was fixed in a recent update.
